packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import {
+	CheckboxControl,
+	__experimentalUseNavigator as useNavigator,
+	Button,
+	Icon,
+} from '@wordpress/components';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { header, footer, layout, chevronRightSmall } from '@wordpress/icons';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SidebarNavigationScreenDetailsPanel,
+	SidebarNavigationScreenDetailsPanelRow,
+	SidebarNavigationScreenDetailsPanelLabel,
+	SidebarNavigationScreenDetailsPanelValue,
+} from '../sidebar-navigation-screen-details-panel';
+import useEditedEntityRecord from '../use-edited-entity-record';
+
+function TemplateAreaButton( { icon, label } ) {
+	const icons = {
+		header,
+		footer,
+	};
+	return (
+		<Button
+			as="a"
+			className="edit-site-sidebar-navigation-screen-template__template-area-button"
+			icon={ icons[ icon ] ?? layout }
+		>
+			{ label }
+			<Icon icon={ chevronRightSmall } />
+		</Button>
+	);
+}
+
+export default function HomeTemplateDetails() {
+	const navigator = useNavigator();
+	const {
+		params: { postType, postId },
+	} = navigator;
+	const { record } = useEditedEntityRecord( postType, postId );
+	const {
+		commentOrder,
+		templatePartAreas,
+		isHomePageBlog,
+		postsPerPage,
+		siteTitle,
+	} = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		const { getSettings } = select( blockEditorStore );
+		const blockEditorSettings = getSettings();
+
+		return {
+			isHomePageBlog:
+				siteSettings?.page_for_posts === siteSettings?.page_on_front,
+			siteTitle: siteSettings?.title,
+			postsPerPage: +siteSettings?.posts_per_page,
```

Comment:
Just a drive-by comment: what does the plus here do? Since optional chaining is used, what happens if siteSettings is undefined, but the plus is applied beforehand... would that result in NaN?

Reply:
It coerces to a number, but there is a risk it will result in a `NaN`. I was counting on a `null` value, but didn't think much of it to be honest! I'll remove since it's a bit too clever. 

Thanks for flagging it 👍